clang_delta/EmptyStructToInt.cpp

```diff
@@ -213,6 +261,58 @@ void EmptyStructToInt::HandleTranslationUnit(ASTContext &Ctx)
     TransError = TransInternalError;
 }
 
+void EmptyStructToInt::handleOneRecordDecl(const RecordDecl *RD,
+                                           const RecordDecl *BaseRD,
+                                           const FieldDecl *FD,
+                                           unsigned int Idx)
+{
+    // Skip field if it is not ofthe type of one of the structs that are affected by the change
+    // Works recursively through chained structs
+    if(!RecordDeclToField[BaseRD] && BaseRD != TheRecordDecl)
+    {
+        return;
+    }
+
+    IndexVector *NewIdxVec = RecordDeclToField[RD];
+
+    if(!NewIdxVec)
+    {
+        NewIdxVec = new IndexVector();
+        RecordDeclToField[RD] = NewIdxVec;
+    }
+
+    NewIdxVec->push_back(Idx);
+}
+
+void EmptyStructToInt::handleOneVarDecl(const VarDecl *VD)
+{
+    const Type *Ty = VD->getType().getTypePtr();
+    const RecordDecl *RD = getBaseRecordDef(Ty);
+
+    // Skip all variables which are not of struct type
+    if(!RD)
+    {
+        return;
+    }
+
+    IndexVector *IdxVec = RecordDeclToField[RD];
+
+    // Skip variables for which the struct is not changed
+    if(!IdxVec && RD != TheRecordDecl)
+    {
+        return;
+    }
+
+    ExprVector InitExprs;
+
+    getInitExprs(Ty, VD->getInit(), IdxVec, InitExprs);
```

Comment:
My apologizes for overlooking the invocation here. Please skip my previous comments.